### ISSUE TITLE
drop (now unneeded) support for subdomains in contributions url

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#19005]
At the time the[ contributions url was added](https://github.com/mozilla/addons-server/issues/6070) there was a specific need for subdomains, but that's no longer the case.

**Comment by AlexandraMoga** [comment]
This issue is verified fixed on stage (also left results in https://bugzilla.mozilla.org/show_bug.cgi?id=1760296):

[domain validation}
![image](https://user-images.githubusercontent.com/31961530/159691120-725c15ff-2720-4a33-8313-139e6a559660.png)


[URL validation for `https://`]
![image](https://user-images.githubusercontent.com/31961530/159691161-3c86d2fe-a2d6-437f-a4f1-d849034012c9.png)

**Comment by eviljeff** [comment]
fwiw, the same fixes were made in the API too (separately - the implementation in code is repeated) but I guess the API change was validated along with #19000

**Comment by AlexandraMoga** [comment]
Yes, I've actually verified in parallel with Devhub and the API, but forgot to mention this.
